### ISSUE TITLE
LazyTextbookDiscretePost perfomance got worse

### ISSUE THREAD

**Issue by kostakoida** [JuliaReach/Reachability.jl#338]
Previously we had:
```
TextbookDiscretePost: 4.488959 seconds (5.85 M allocations: 313.031 MiB, 7.18% gc time)
 LazyTextbookDiscretePost: 1.354744 seconds (2.52 M allocations: 172.287 MiB, 20.40% gc time)
Overaprroximation:  1.020842 seconds (3.95 M allocations: 312.647 MiB, 19.77% gc time)
```
Now we have:
```
TextbookDiscretePost : 1.301113 seconds (1.81 M allocations: 89.292 MiB, 8.14% gc time)
LazyTextbookDiscretePost:  32.640685 seconds (77.95 M allocations: 4.571 GiB, 5.84% gc time)
Overaprroximation: 0.470141 seconds (1.44 M allocations: 127.239 MiB, 16.40% gc time)
```

I have been on: 
```
Merge pull request #321 from JuliaReach/schillic/319
#319 - Depth limit for line search

Merge pull request #819 from JuliaReach/schillic/817
#817 - Add method to swap arguments in a lazy operation
```
for Reachability.jl and LazySets.jl respectively.

**Comment by mforets** [comment]
Could you please update timings in "Now we have:" wrt current master branches?

**Comment by mforets** [comment]
there are 3 places where `LazyTextbookDiscretePost` differs from `ApproximatingTextbookDiscretePost`, 

- in `tube⋂inv!`  for the intersection with the invariant, `R⋂I`
- two in `post`, first with the guard, then with the target invariant

proposal: make `LazyTextbookDiscretePost`  less lazy, by taking the overapprox of `R⋂I`.

**Comment by schillic** [comment]
> proposal: make LazyTextbookDiscretePost less lazy, by taking the overapprox of R⋂I.

Okay, but I would make this optional, and it does not belong here (create a new issue). I imagine that for property checking, keeping the intersection lazy will be very helpful.

**Comment by schillic** [comment]
This was fixed in #345 and #347.
